src/m365/spo/commands/site/site-apppermission-set.ts

```diff
@@ -0,0 +1,169 @@
+import { Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import GraphCommand from '../../../base/GraphCommand';
+import SpoCommand from '../../../base/SpoCommand';
+import commands from '../../commands';
+import { SitePermission, SitePermissionIdentitySet } from './SitePermission';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  siteUrl: string;
+  permissionId?: string;
+  appId?: string;
+  appDisplayName?: string;
+}
+
+class SpoSiteAppPermissionSetCommand extends GraphCommand {
+  private siteId: string = '';
+
+  public get name(): string {
+    return commands.SITE_APPPERMISSION_SET;
+  }
+
+  public get description(): string {
+    return 'Updates a specific application permission for a site';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.permissionId = typeof args.options.permissionId !== 'undefined';
+    telemetryProps.appId = typeof args.options.appId !== 'undefined';
+    telemetryProps.appDisplayName = typeof args.options.appDisplayName !== 'undefined';
+    return telemetryProps;
+  }
+
+  private getSpoSiteId(args: CommandArgs): Promise<string> {
+    const url = new URL(args.options.siteUrl);
+    const siteRequestOptions: any = {
+      url: `${this.resource}/v1.0/sites/${url.hostname}:${url.pathname}`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ id: string }>(siteRequestOptions)
+      .then((site: { id: string }) => site.id);
+  }
+
+  private getFilteredPermissions(args: CommandArgs, permissions: SitePermission[]): SitePermission[] {
+    let filterProperty: string = 'displayName';
+    let filterValue: string = args.options.appDisplayName as string;
+
+    if (args.options.appId) {
+      filterProperty = 'id';
+      filterValue = args.options.appId;
+    }
+
+    return permissions.filter((p: SitePermission) => {
+      return p.grantedToIdentities.some(({ application }: SitePermissionIdentitySet) =>
```

Comment:
We can skip the `return` keyword here and shorten it to `=> p.grantedToIdentities...`